samples/js/text_generation/compound_grammar_generation.js

```diff
@@ -97,34 +120,38 @@ async function main() {
     const userText1 = "Do dolphins have fingers?";
     console.log("User: ", userText1);
     chatHistory.push({ role: "user", content: userText1 });
-    const modelInput = tokenizer.applyChatTemplate(chatHistory, true);
+    const modelInput = tokenizer.applyChatTemplate(chatHistory, true, undefined, tools);
```

Comment:
If tools are constant, we could set `tools` to `chatHistory` directly with `chatHistory.setTools(tools)` to setup default tools behavior for this history. In that case we will not need to apply chat template with `tokenizer.applyChatTemplate()` manually in each conversation iteration.